fairlearn/adversarial/_preprocessor.py

```diff
@@ -103,11 +103,16 @@ def _check(self, X, dtype=None, init=False):
         )
         if init:
             self.input_dim_ = X.ndim
+            if self.input_dim_ > 2:
+                raise ValueError(
+                    "Data can be at most two dimensional, not %d" % self.input_dim_
+                )
         else:
             if X.ndim != self.input_dim_:
                 raise ValueError("Dimension of data is inconsistent with previous call")
         if X.ndim == 1:
             X = X.reshape(-1, 1)
+        assert X.ndim == 2
```

Comment:
An if check with raising an exception with a nice error message is probably preferable. Unless this is a case that can't happen at all?

Reply:
It really shouldn't happen, but I agree!